.travis.yml

```diff
@@ -8,6 +8,8 @@ addons:
     - ubuntu-toolchain-r-test
     packages:
     - g++-4.8
+services:
+  - redis-server
```

Comment:
It seems slightly weird that the tests would now require that we have a redis up and running on a specific port, I liked @erinspice's refactor that made the test-suite stand up the testing server for this purpose.

Perhaps we could make the tests stand up a server on `:6379`, but only do so if there's not already a server running on `:6379`? I'm happy to make this change myself when I'm done the work day.